src/analyze/analysis_handler.go

```diff
@@ -6,85 +6,148 @@ import (
+			}
+		}
 
-		ShakeAlertOutliers:   shakeAlertOutliers,
-		ShakeAlertTimestamps: shakeAlertOutlierTime,
+		// Update outliers for both BLT MAD and ShakeAlert outliers
+		updateOutliers(AllResults, BLTMAD, bltIndexes, sortedTimestamps, sortedFrequencies)
+		updateOutliers(AllResults, SHAKEALERT, shakeAlertIndexes, sortedTimestamps, sortedFrequencies)
 	}
 
-	//what is the best way to output this?
-	m := common.OutlierMessages{
-		MADOutlierMessages: bltOutlierMessages,
-		ShakeAlertMessages: shakeAlertOutlierMessages}
-
-	blt_mad.StoreResultIntoJson(r, recentResultsFile) //storing the most recent result
-	blt_mad.WriteCSVFile(m, outlierMessagesFile)
-	//make sure that we do not get more than a threshold number of points
-	if len(AllResults)+1 > maxPoints {
-		AllResults = append(AllResults[1:], r) //append all the elements except for the first one
-	} else {
-		AllResults = append(AllResults, r)
+}
+
+func getListOfKeys(dataMap map[time.Time]float64) []time.Time {
+	keys := []time.Time{}
+	for timestamp, _ := range dataMap {
+		keys = append(keys, timestamp)
 	}
-	return r
+	return keys
 }
 
-//changed bltMad inputs to get timestamps and the outliers at the same time
-//this is also producing duplicates
-func BltMadWindow(data []float64, window common.Window, tau float64) ([]float64, []time.Time, [][]common.BGPMessage) {
-	var outliers []float64
-	var times []time.Time
-	var messages [][]common.BGPMessage //array of arrays of messages for a given bucket map
-	bucketMap := window.BucketMap
+//use the olderst timestamp when removing the data points
+func getSmallestTimestamp(timestamps []time.Time) time.Time {
+	if len(timestamps) == 0 {
+		return time.Time{}
+	}
 
-	lengthMap := makeLengthMap(window)
+	smallest := timestamps[0]
+	for _, ts := range timestamps {
+		if ts.Before(smallest) {
+			smallest = ts
+		}
+	}
+	return smallest
+}
 
-	for timestamp, _ := range lengthMap {
-		if blt_mad.IsAnOutlierBLT(data, tau, lengthMap[timestamp]) {
-			outliers = append(outliers, lengthMap[timestamp])
-			times = append(times, timestamp)
-			messages = append(messages, bucketMap[timestamp])
+func updateOutliers(currResult common.Result, algorithm int, indexes []int, sortedTimestamps []time.Time, sortedFrequencies []float64) {
+	for _, index := range indexes {
+		// Get outlier timestamp
+		outlierTimestamp := sortedTimestamps[index]
+		outlierFrequency := sortedFrequencies[index]
+
+		// Go through current outliers and see if timestamp exists
+		currOutlier, exists := currResult.AllOutliers[outlierTimestamp]
+
+		// If the outlier is not already in the map, create a new outlier for it
+		if !exists {
+			currResult.AllOutliers[outlierTimestamp] = createOutlierStruct(outlierTimestamp, algorithm, outlierFrequency)
+		} else { // If outlier not in map, need to possibly update it
+			if currOutlier.Count != outlierFrequency {
+				currOutlier.Count = outlierFrequency
+			}
+			// Update algorithm based on conditions
+			switch {
```

Comment:
Lets avoid nesting if statements inside switch blocks.

Reply:
^addressing in the next pr